relayer/workers/parachaincommitmentrelayer/catchup.go

```diff
@@ -155,11 +157,31 @@ func (li *BeefyListener) parablocksWithProofs(blocks []ParaBlockWithDigest, late
 				return nil, err
 			}
 			li.log.WithField("relayBlockHash", relayBlockHash.Hex()).Info("Got relay chain blockhash")
-			allParaHeads, ownParaHeadProofPos, ownParaHead, err = li.relaychainConn.GetAllParaheadsWithOwn(relayBlockHash, li.paraID)
+			heads, err := li.relaychainConn.FetchParaHeads(relayBlockHash)
 			if err != nil {
 				li.log.WithError(err).Error("Failed to get paraheads")
 				return nil, err
 			}
+
+			if _, ok := heads[li.paraID]; !ok {
+				return nil, fmt.Errorf("chain is not a registered parachain")
+			}
+
+			var header types.Header
+			if err := types.DecodeFromBytes(heads[li.paraID].Data, &header); err != nil {
+				li.log.WithError(err).Error("Failed to decode Header")
+				return nil, err
+			}
+
+			tmp := make([]types.Bytes, 0, len(heads))
+			for _, v := range heads {
```

Comment:
does this guarantee ordering, ie, guarantee that the order of heads in the byte array will be ascending based on LeadIndex?

Reply:
ah good catch. Will need to order by LeafIndex.

Reply:
I've come up with the following remedy:

* `FetchParaHeads` remains unchanged:
* Added a method `AsProofInput` which transforms the map of heads into a form suitable for proof generation

I think its fine to transition from an ordered array to map and then back to array if it allows us to break down complexity into manageable, composable parts. Its kinda like a data pipeline. The extra overhead of this pipeline and sorting is minimal since N (number of parachains) is gonna be less than 200 which is practically O(1) for a computer.

Reply:
yep agree, awesome thx